client/ayon_deadline/addon.py

```diff
@@ -81,6 +91,157 @@ def get_server_info_by_name(self, server_name: str) -> DeadlineServerInfo:
 
         return server_info
 
+    def submit_job(
+        self,
+        server_name: str,
+        plugin_info: Dict[str, Any],
+        job_info: Dict[str, Any],
+        aux_files: Optional[List[str]] = None,
+    ) -> Dict[str, Any]:
+        """Submit job to Deadline.
+
+        Args:
+            server_name (str): Deadline Server name from project Settings.
+            plugin_info (dict): Plugin info data.
+            job_info (dict): Job info data.
+            aux_files (Optional[List[str]]): List of auxiliary files.
+
+        Returns:
+            str: Job ID.
+
+        """
+        payload = {
+            "JobInfo": job_info,
+            "PluginInfo": plugin_info,
+            "AuxFiles": aux_files or [],
+        }
+        server_url, auth, verify = self._get_deadline_con_info(server_name)
+        response = requests.post(
+            f"{server_url}/api/jobs",
+            json=payload,
+            timeout=10,
+            auth=auth,
+            verify=verify
+        )
+        if not response.ok:
+            raise ValueError("Failed to create job")
+
+        payload["job_id"] = response.json()["_id"]
+        return payload
```

Comment:
Should we separate this further - so that we just have a `submit` or `get` and `post` call on the addon or alike so if someone wants to just do any sort of query against the deadline web service they are free to do so using the API of the addon instead of having to figure out the relevant `auth`,  etc.?

Reply:
I would not make `get` and `post`, rather more targeted methods we actually need. As far as I know for most of what we use we already have them, e.g. we already do have method to collect server info (pools etc.). Do we need anything else?

Reply:
> I would not make `get` and `post`, rather more targeted methods we actually need. As far as I know for most of what we use we already have them, e.g. we already do have method to collect server info (pools etc.). Do we need anything else?
> 
> But we might make `_get_deadline_con_info` public so it can be used elsewhere if needed...

Not necessarily that I can think of now - but what if another addon wanted to also manage says the available pools on Deadline, and "change" that - or list all jobs, etc. Not a big issue - but it would've made that easily accessible.

However, I think with `get_deadline_server_connection_info` now exposed it's relatively light-weight to do any type of query, even without using `requests` but some other library if needed. So 👍